httptransport/logginghandler.go

```diff
@@ -0,0 +1,34 @@
+package httptransport
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+type httpStatusWriter struct {
+	http.ResponseWriter
+
+	StatusCode int
+}
+
+// LoggingHandler will log HTTP requests using the pre initialized zerolog.
+func LoggingHandler(next http.Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		start := time.Now()
+
+		log := zerolog.Ctx(r.Context())
+
+		next.ServeHTTP(w, r)
+
+		log.Info().
+			Str("remote addr", r.RemoteAddr).
+			Str("method", r.Method).
+			Str("request uri", r.RequestURI).
+			Str("status", strconv.Itoa(http.StatusOK)).
+			Float64("elapsed time (md)", float64(time.Since(start).Nanoseconds())*1e-6).
```

Comment:
small nit; dividing by the constants in time package maybe a little cleaner.

```
(time.Since(start).Nanoseconds()/int64(time.Millisecond))
```

Reply:
Also, does this need to be a float64 at all? Looks like you get an Int with Nanoseconds.

Reply:
Makes sense, only thing here is you might loose that visibility moving to an int64, i.e. you wont see 0.013 instead will be just 0 milliseconds. But def cleaner and easier to follow.

Reply:
Ah, actually missed the second float64 wrap. I see now. Your call.